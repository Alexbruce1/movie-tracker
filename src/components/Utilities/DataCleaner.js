import * as API from './API'

export const fetchMovies = async () => {
	const apiKey = '26d5b93e45b773596adda2d2b99efa0f'
	const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${apiKey}&language=en-US&page=1`

	const movieData = await API.fetchData(url)
	const currentMovies = await returnMovieData(movieData.results)
	return currentMovies
} 

export const returnMovieData = async (movies) => {
	const moviePromises = movies.map( async movie => {

		return {
			title: movie.title,
			poster: await fetchMoviePoster(movie),
			overview: movie.overview,
			releaseDate: formatReleaseDate(movie.release_date),
			rating: movie.vote_average
		}
	})
	return Promise.all(moviePromises)
}

export const fetchMoviePoster = async (movie) => {
	const apiKey = '26d5b93e45b773596adda2d2b99efa0f'
	const posterUrl = `https://api.themoviedb.org/3/movie/${movie.id}/images?api_key=${apiKey}&language=en-US`
	const fetchedPoster = await API.fetchImage(posterUrl)
	return fetchedPoster.posters
}

const formatReleaseDate = (date) => {
	const month = date.slice(5,7)
	const day = date.slice(8,10)
	const year = date.slice(0,4)
	return `${month}/${day}/${year}`
}