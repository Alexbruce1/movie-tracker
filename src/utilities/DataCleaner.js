import * as API from './API'

export const fetchMovies = async () => {
	const apiKey = '26d5b93e45b773596adda2d2b99efa0f'
	const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${apiKey}&language=en-US&page=1`

	const movieData = await API.fetchData(url)
	const currentMovies = await returnMovieData(movieData.results)
	return currentMovies
}

export const returnMovieData = async (movies) => {
	const moviePromises = movies.map( async movie => {

		return {
			title: movie.title,
			poster: `https://image.tmdb.org/t/p/w500/${movie.poster_path}`,
			overview: movie.overview,
			releaseDate: formatReleaseDate(movie.release_date),
			rating: movie.vote_average
		}
	})
	return Promise.all(moviePromises)
}

const formatReleaseDate = (date) => {
	const month = date.slice(5,7)
	const day = date.slice(8,10)
	const year = date.slice(0,4)
	return `${month}/${day}/${year}`
}