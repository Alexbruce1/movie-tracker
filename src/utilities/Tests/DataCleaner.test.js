import * as DataCleaner from '../DataCleaner';

describe('Data Cleaner', () => {
  describe('fetchMovies', () => {
  });

  describe('returnMovieData', () => {
  });

  describe('formatReleaseDate', () => {
    it('Should format a release date', () => {
      const mockReleaseDate = '2018-10-03';
      const expected = '10/03/2018';
      const result = DataCleaner.formatReleaseDate(mockReleaseDate);
      expect(result).toBe(expected);
    });
  });
});
