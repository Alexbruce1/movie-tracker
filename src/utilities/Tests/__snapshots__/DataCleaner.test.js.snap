// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Cleaner returnMovieData Should clean movie objects 1`] = `
Array [
  Object {
    "backdrop": "https://image.tmdb.org/t/p/w500/url",
    "favorite": false,
    "id": 112432,
    "overview": "",
    "poster": "https://image.tmdb.org/t/p/w500/url",
    "rating": 5,
    "releaseDate": "05/13/2018",
    "title": "A movie",
  },
  Object {
    "backdrop": "https://image.tmdb.org/t/p/w500/url",
    "favorite": false,
    "id": 11423,
    "overview": "",
    "poster": "https://image.tmdb.org/t/p/w500/url",
    "rating": undefined,
    "releaseDate": "07/08/2018",
    "title": "Another movie",
  },
]
`;
